recording/lib/storage/s3.ex

```diff
@@ -132,44 +113,76 @@ defmodule Membrane.RTC.Engine.Endpoint.Recording.Storage.S3 do
 
     case response do
       {:ok, %{body: %{contents: contents}}} ->
-        objects =
-          Enum.map(contents, fn stats ->
-            filename = stats.key |> String.split("/") |> List.last()
-            size = String.to_integer(stats.size)
-            {filename, size}
-          end)
-          |> Enum.into(%{})
-
-        {:ok, objects}
+        {:ok, Map.new(contents, &parse_stats/1)}
 
       _else ->
+        Membrane.Logger.error("Couldn't list objects on S3 bucket, recording id: #{recording_id}")
         {:error, :list_objects}
     end
   end
 
-  defp clean_objects(objects, %{credentials: credentials}) do
+  defp clean_objects(objects, recording_id, %{credentials: credentials}) do
     config = create_aws_config(credentials)
 
-    credentials.bucket
-    |> ExAws.S3.delete_all_objects(objects)
-    |> ExAws.request(config)
-  end
+    result =
+      credentials.bucket
+      |> ExAws.S3.delete_all_objects(objects)
+      |> ExAws.request(config)
+
+    case result do
+      {:ok, _term} ->
+        :ok
 
-  defp handle_error({:error, :list_objects}, options) do
-    Membrane.Logger.error(
-      "Couldn't list objects on S3 bucket, recording id: #{options.recording_id}"
-    )
+      {:error, _reason} ->
+        Membrane.Logger.error(
+          "Couldn't clean objects on S3 bucket, recording id: #{recording_id}"
+        )
 
-    :error
+        :error
+    end
   end
 
-  defp handle_error({:error, :failed_to_clean_objects}, options) do
-    Membrane.Logger.error(
-      "Couldn't clean objects on S3 bucket, recording id: #{options.recording_id}"
-    )
+  defp fix_object(
+         {:ok, s3_size},
+         file_size,
+         _file_path,
+         _filename,
+         _recording_id,
+         _storage_opts
+       )
+       when s3_size >= file_size,
+       do: true
+
+  defp fix_object(
+         _s3_size,
+         _file_size,
+         file_path,
+         filename,
+         recording_id,
+         storage_opts
+       ) do
+    config =
+      file_path
+      |> File.read!()
+      |> save_object_config(recording_id, filename)
+
+    case save_object(config, storage_opts) do
+      :ok -> true
+      {:error, _response} -> false
+    end
```

Comment:
thought: This looks like two functions merged into one, they even don't share parameters.
What do you think about splitting them up and being more verbose? Functions names could be: `same_file?` and `fix_s3_file?`.